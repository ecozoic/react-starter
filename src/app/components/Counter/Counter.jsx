// @flow
/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

type Props = {
  +count: number,
  +onIncrement: Function,
  +onDecrement: Function,
};

/**
 * Component to render a counter
 * @param {Props} props
 * @param {Number} props.count - the current counter value
 * @param {Function} props.onIncrement - function to call when increment button is clicked
 * @param {Function} props.onDecrement - function to call when decrement button is clicked
 */
export const Counter = ({ count, onIncrement, onDecrement }: Props) => (
  <div>
    <div>{count}</div>
    <button onClick={onDecrement}>-</button>
    <button onClick={onIncrement}>+</button>
    <div>
      <Link to="/admin">ADMIN AREA</Link>
    </div>
  </div>
);

Counter.propTypes = {
  count: PropTypes.number.isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};

export default connect(
  state => ({ count: state.counter.count }),
  dispatch => ({
    onIncrement: () => dispatch({ type: 'INCREMENT' }),
    onDecrement: () => dispatch({ type: 'DECREMENT' }),
  }),
)(Counter);
