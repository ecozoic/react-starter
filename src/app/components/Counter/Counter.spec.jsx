import React from 'react';
import { shallow } from 'enzyme';

import { Counter } from './Counter';

describe('Counter', () => {
  it('renders', () => {
    const noop = () => undefined;

    // eslint-disable-next-line function-paren-newline
    const counter = shallow(
      <Counter
        count={0}
        onIncrement={noop}
        onDecrement={noop}
      />,
    // eslint-disable-next-line function-paren-newline
    );

    expect(counter).toMatchSnapshot();
  });
});
