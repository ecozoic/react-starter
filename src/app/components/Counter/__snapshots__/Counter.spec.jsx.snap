// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders 1`] = `
<div>
  <div>
    0
  </div>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
